### PR TITLE
Fix bug in AlarmClock.listAlarms()

### PR DESCRIPTION
**Pull request by ebaauw**
Fix for #211, where the return of `AlarmClock.ListAlarms()` is wrong.

Not sure what will be impacted by this change.  Alternatively, we could pass an optional parameter to `Service._parseKey()` for `explicitArray`, so each caller has their own say in this.

### CODE REVIEW

**Comment by ebaauw** [comment]
No this is not good.  Now `_parseKey()` creates arrays for all fields.

**Comment by ebaauw** [comment]
At least, for take 2 the impact is limited to (the users of) `listAlarms()`.  This seems to do what I want it to do, but I'm not sure if this is the proper way.  Is there any reason to test `typeof ... === 'undefined'` instead of `... === undefined`?

File: lib/services/Service.js [hunk 1]
```diff
@@ -94,7 +94,7 @@ Service.prototype._request = function (action, variables, callback) {
  * @return {void}
  */
 Service.prototype._parseKey = function (input, key, callback) {
-  debug('Service._parseKey(%j, %j)', key, callback)
+  debug('Service._parseKey(%j, %j, %j)', input, key, callback)
```
Comment (svrooij) [inline lib/services/Service.js]: This line makes it very verbose. Can you put it in as a comment, for others to use?

Comment (ebaauw) [reply lib/services/Service.js]: Sure

**Pull request merged**